### PR TITLE
Task 35 routes guide

### LINKED ISSUE #35: Write guide for Routes demo app

### PR DESCRIPTION
**Pull request by rajinder-yadav**
resolves #35

### CODE REVIEW

**Comment by rajinder-yadav** [comment]
tracks #35

**Comment by stevenkampen** [comment]
LGTM :shipit:

File: guide/routes/routes.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+The _city_ module `city.module.ts` contains:
+
+* CityComponent
+* CityOneComponent
+* CityTwoComponent
+* CityThreeComponent
+
+## Architecture
+
+The component code is simple, the `AppComponent` has a single router outlet. When the `ParksComponent` or `CityComponent` component is loaded, a second router outlet become available. Components from the child route get loaded into the second router outlet.
```
Comment (tyllrb) [inline guide/routes/routes.md]: ...a second router outlet **becomes** available.

File: guide/routes/routes.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+Routes|Component (View)
+------|---------
+`/`|AppComponent
+`/home`|HomeComponent
+`/parks`|ParksComponent
+`/parks/park1`|ParkOneComponent
+`/parks/park2`|ParkTwoComponent
+`/parks/park3`|ParkThreeComponent
+
+The root path `/` gets redirect to `/home`. Also the lazy loaded route for `cities` is defined in the _root_ module.
```
Comment (tyllrb) [inline guide/routes/routes.md]: ...gets **redirected** to `/home`.

File: guide/routes/routes.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+* Park 1
+* Path 2
+* Park 3
+
+Notice how each child component is loaded under `ParksComponent` just underneath `router-outlet`. This is how Angular inserts a component into the DOM, as a sibliing element of `router-outlet`.
+
+## Routes
+
+To see the current _routes_ for the application, click on the _Router Tree_ tab. This will show all the defined routes that are currently loaded in the application. The routes shows the path to each component (think view). The `AppComponent` path is the root `/`. The path to `HomeComponent` is `/home`, likewise the path to the a child component `ParkThreeComponent` is `/parks/park3`.
```
Comment (tyllrb) [inline guide/routes/routes.md]: ...likewise the path to **the** child component...

File: guide/routes/routes.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+![Image Route tree](images/router-tree.png)
+
+Take a look at the bottom of the router tree, you will notice `cities [Lazy]`, this is a lazy loaded route. This means that the child components for the path `/cities` will be loaded on demand, this takes place when the _Cities_ button is clicked. Let's see this in action, click on the _Cities_ button and pay attention to how the _lazy route_ is updated when the feature Angular module is loaded.
+
+![Image Lazy routes](images/routes-lazy.gif)
```
Comment (tyllrb) [inline guide/routes/routes.md]: Does the blue colour of the park components mean something different than the red colour of the city components?

**Comment by tyllrb** [comment]
Looks good, just a few spelling fixes and one small question about the colours in the router tree.

File: guide/routes/routes.md (hunk 1 above)
Comment (rajinder-yadav) [inline guide/routes/routes.md]: fixed

File: guide/routes/routes.md (hunk 2 above)
Comment (rajinder-yadav) [inline guide/routes/routes.md]: fixed

File: guide/routes/routes.md (hunk 3 above)
Comment (rajinder-yadav) [inline guide/routes/routes.md]: fixed

File: guide/routes/routes.md (hunk 4 above)
Comment (rajinder-yadav) [inline guide/routes/routes.md]: Colours are already covered in the IO Demo.

**Pull request merged**